compaction/time_window_compaction_strategy.cc

```diff
@@ -401,7 +401,7 @@ void time_window_compaction_strategy::update_estimated_compaction_by_tasks(std::
             n += size_tiered_compaction_strategy::estimated_pending_compactions(bucket, min_threshold, max_threshold, _stcs_options);
             break;
         case bucket_compaction_mode::major:
-            n++;
+            n += bucket.size() > 1;
         default:
```

Comment:
For a bucket to fall into major mode, this is the condition:
```
    if (bucket.size() >= 2 && !is_last_active_bucket(bucket_key, now) && _recent_active_windows.contains(bucket_key)) {
        return bucket_compaction_mode::major;
```

A bucket with a single file will fall into none condition, meaning it doesn't have to be compacted.

Reply:
I see, so although the patch will work it is not needed.

Reply:
dequeued

Reply:
Thanks!